onnx/onnx.proto

```diff
@@ -237,6 +245,172 @@ message TensorProto {
 message SparseTensorProto {
   // The dimensions in the tensor.
   repeated int64 dims = 1;
+  // This field MUST be present this version of the IR.
   optional TensorProto indices = 2;
+  // This field MUST be present this version of the IR.  
   optional TensorProto values = 3;
 }
+
+// A ValueProto represents a value that may be serialized into a model.
+message ValueProto {
+
+  message KeyValuePairProto {
+    oneof key {
+      string s = 1;
+      int32 i32 = 2;
+      int64 i64 = 3;
+      uint64 ui64 = 4;
+    };
+    optional ValueProto value = 100;
+  }
+
+  message NameValuePairProto {
+    optional string key = 1;
+    optional ValueProto value = 100;
+  }
+
+  // Defines a record in its serialized format.
+  // A record is a sequence of one or more distinctly
+  // typed uniquely named values. 
+  message RecordProto {
+    repeated NameValuePairProto fields = 1;
+  }
+
+  // Defines a union value  in its serialized format.
+  // The name is the discriminator from the union type.
+  message UnionProto {
+    optional NamedValueProto choice = 1;
```

Comment:
Is that a NameValuePairProto? Didn't see a NamedValueProto ... maybe I missed it.